SDK: Import notifications panel code into Calypso Take 3

dmsnell opened the pull request:
See #26524, a previous attempt

In this patch I'm pulling in the code from
Automattic/notifications-panel without making any deeper changes in
Calypso. As a preparatory change it should allow us to focus on the bits
in later commits which _do_ change the system as a whole.

This is the first of several patches meant to integrate the
notifications panel into Calypso and eliminate the separate repository
and build system. We hope to increase the visibility to the
notifications panel to other developers, improve the release efficiency,
and start allowing for deeper code sharing by doing this.

We're merging with `eslint` warnings and errors disabled because we are
importing a large body of existing code that was developed with
different coding standards.

After this merges we should have the ability to import the notifications
panel from this directory directly instead of through an awkward import
from `node_modules`.

We should move fast after this to remove the existing repository and
connect all the important pieces to prevent the confusion of having the
code in both places - we don't want the code to get out of sync.

**Testing**

Build Calypso and make sure this doesn't affect the output. We should have
no change in functionality, visuals, or output sizes (or content).

tyxla reviewed (approved):
LGTM 🚢 

Let's ship this as-is, and execute any further work in subsequent PRs to keep them contained and easy to iterate.

dmsnell commented:
Note for the curious passerby: we're merging with style errors here because…
 - we're merging in an existing large codebase
 - we're not importing this new code yet and so it should't ever make contact with the main Calypso build until a later PR
 - `npm run test` succeeded locally without any errors
 - we're confirming on staging that things work

dmsnell committed:
SDK: Import notifications panel code into Calypso

See #26524, a previous attempt

In this patch I'm pulling in the code from
Automattic/notifications-panel without making any deeper changes in
Calypso. As a preparatory change it should allow us to focus on the bits
in later commits which _do_ change the system as a whole.

This is the first of several patches meant to integrate the
notifications panel into Calypso and eliminate the separate repository
and build system. We hope to increase the visibility to the
notifications panel to other developers, improve the release efficiency,
and start allowing for deeper code sharing by doing this.

We're merging with `eslint` warnings and errors disabled because we are
importing a large body of existing code that was developed with
different coding standards.

After this merges we should have the ability to import the notifications
panel from this directory directly instead of through an awkward import
from `node_modules`.

We should move fast after this to remove the existing repository and
connect all the important pieces to prevent the confusion of having the
code in both places - we don't want the code to get out of sync.

**Testing**

Build Calypso and make sure this doesn't affect the output. We should have
no change in functionality, visuals, or output sizes (or content).
